### PR TITLE
feat: Node pools can enable fast_socket

### LINKED ISSUE #2199: Node pools can enable fast sockets
### TL;DR

A node_pool should be able to enable `fast_socket` in its `node_config`

### Terraform Resources

The `node_config` has [fast_socket](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#fast_socket-1)

### Detailed design

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by fraenkel**
fixes #2199

### CODE REVIEW

File: autogen/main/cluster.tf.tmpl [hunk 1]
```diff
@@ -537,6 +537,13 @@ resource "google_container_cluster" "primary" {
         }
       }
 
+      dynamic "fast_socket" {
+        for_each = lookup(var.node_pools[0], "enable_fast_socket", false) ? [true] : []
```
Comment (apeabody) [inline autogen/main/cluster.tf.tmpl:541]: I recommend defaulting to `null`, that way the content value can be populated for both `true` and `false`.
```suggestion
        for_each = lookup(var.node_pools[0], "enable_fast_socket", null) != null ? [var.node_pools[0].enable_fast_socket] : []
```

File: autogen/main/cluster.tf.tmpl [hunk 2]
```diff
@@ -930,6 +937,12 @@ resource "google_container_node_pool" "windows_pools" {
         enabled = gvnic.value
       }
     }
+    dynamic "fast_socket" {
+      for_each = lookup(each.value, "enable_fast_socket", false) ? [true] : []
```
Comment (apeabody) [inline autogen/main/cluster.tf.tmpl:941]: same here

**Review by apeabody — CHANGES_REQUESTED** [review]
Thanks for the contribution @fraenkel!

File: autogen/main/cluster.tf.tmpl (hunk 1 above)
Comment (fraenkel) [reply autogen/main/cluster.tf.tmpl]: fixed

**Comment by apeabody** [comment]
Thanks @fraenkel - The Terraform looks good to me, please also add the new field to the node pool documentation here: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/autogen/main/README.md?plain=1#L186

**Comment by fraenkel** [comment]
The READMEs have been updated.

**Comment by apeabody** [comment]
validated `fast_socket` is present in v6.10

**Review by apeabody — APPROVED** [review]

**Pull request merged**